modules/ctl/src/main/scala/free/interp.scala

```diff
@@ -60,6 +60,7 @@ object interpreter {
    * Construct a program to log a message to the console at the given log level and indentation.
    * This is where all the colorizing happens.
    */
+  @SuppressWarnings(Array("org.wartremover.warts.ToString"))
```

Comment:
Wartremover triggers a compile error if you use `.toString` on any type that doesn't override the default implementation. So we need to change `ProgramId` to override `toString` rather than overriding it in all of the case classes. This is kind of irritating but I think the wart is reasonable.